collector/mesos_agent/mesos_agent.go

```diff
@@ -115,36 +149,109 @@ type resourceStatistics struct {
 	NetTxDropped uint64 `json:"net_tx_dropped,omitempty"`
 }
 
+func (h *MesosAgentCollector) RunPoller() {
+	ticker := time.NewTicker(h.PollPeriod * time.Second)
+
+	// Poll once immediately
+	h.pollMesosAgent()
+	for _, m := range h.transformContainerMetrics() {
+		h.MetricsChan <- m
+	}
+
+	for {
+		select {
+		case _ = <-ticker.C:
+			h.pollMesosAgent()
+			for _, m := range h.transformContainerMetrics() {
+				h.MetricsChan <- m
+			}
+		}
+	}
+}
+
+func (h *MesosAgentCollector) pollMesosAgent() {
+	now := time.Now().UTC()
+	h.timestamp = now.Unix()
+
+	host := fmt.Sprintf("%s:%d", h.NodeInfo.IPAddress, h.Port)
+
+	// always fetch/emit agent state first: downstream will use it for tagging metrics
+	maColLog.Debugf("Fetching state from DC/OS host %s", host)
+	if err := h.getAgentState(); err != nil {
+		maColLog.Errorf("Failed to get agent state from %s. Error: %s", host, err)
```

Comment:
True, and I considered breaking them into separate funcs. However, we would only print the error and pass, then run the next individual method if we broke them out. Since the behavior is identical in both case, I left it as-is.